src/statefulmodelinstance.hpp

```diff
@@ -20,12 +20,12 @@
 #include <string>
 
 #include "global_sequences_viewer.hpp"
+#include "metric_config.hpp"
+#include "metric_registry.hpp"
```

Comment:
Why we need to add headers here?
I assume cppclean complains here. The reason is most probably constructor definition inside hpp file. Please check if that will resolve this issue.

Both MetricRegistry & MetricConfig  are only passed as reference/pointer here so there should be no need for those headers.